### ISSUE TITLE
Transparent icon for some extensions that supply a DevTools Panel

### ISSUE THREAD

**Issue by bengry** [MicrosoftEdge/DevTools#237]
It seems like some extensions are getting a transparent icon in their devtools panel.
E.g. [EditThisCookie](https://chromewebstore.google.com/detail/editthiscookie/fngmhnnpilhplaeedifhccceomclgfbg)
![CleanShot 2024-03-20 at 20 17 46@2x](https://github.com/MicrosoftEdge/DevTools/assets/260431/2cd750aa-80d6-453f-b111-a48c814763ed)

Trying to debug this a bit it seems like the `<devtools-icon>` element has the correct value for the icon source as it's `[name]`, and in the root `<span>` (inside the shadow root) a wrong `--icon-url` CSS variable is injected. In this case it's:
`url(chrome-extension://fngmhnnpilhplaeedifhccceomclgfbg/img/icon_32x32.png.svg)`, which leads to a file that doesn't exist. I'm not sure why the `.svg` suffix is automatically appended.

Furthermore, if I manually remove the `.svg` suffix, a mask in the same shape as the icon is displayed, instead of the icon:
![CleanShot 2024-03-20 at 20 21 42@2x](https://github.com/MicrosoftEdge/DevTools/assets/260431/59d0ef1d-9b54-4be2-ae5b-065590ed9601)

I think adding the extension's icon is an Edge-specific feature, as in Chrome this is what I get:
![CleanShot 2024-03-20 at 20 23 02@2x](https://github.com/MicrosoftEdge/DevTools/assets/260431/ac14ef5c-3802-48b5-8b38-0587f29b73c9)

OS: macOS 14.4 (23E214)
Edge version: Version 122.0.2365.92

AB#49588478

**Comment by captainbrosset** [comment]
Extension icons have been a bit of a challenge to get right lately. So thank you for filing this issue and investigating already.
I'll make sure our dev team prioritizes this in our backlog.

**Comment by bengry** [comment]
> Extension icons have been a bit of a challenge to get right lately. So thank you for filing this issue and investigating already. I'll make sure our dev team prioritizes this in our backlog.

Thanks @captainbrosset! It would also be nice if we could just turn them off completely. I'm not sure how useful they are, especially when each has its own size and look. Unlike the built-in panels which look & feel uniform.

**Comment by captainbrosset** [comment]
It would be nice if each and every panel had its own icon. This would help make things look consistent, and some people like to use icons to quickly identify tools. But I agree with the risk of them having their own different look.

**Comment by bengry** [comment]
> It would be nice if each and every panel had its own icon. This would help make things look consistent, and some people like to use icons to quickly identify tools. But I agree with the risk of them having their own different look.

I agree in principle, but you can't force the hand of extension developers, especially when most of them think of Edge specially (which is the only browser that has this out of the Chromium browsers I looked at).

That's why I suggested an option to turn off extension icons - so that the consumer has the control.

**Comment by Ilanaya** [comment]
As this is still an open issue, I'll provide some examples of extensions with working/broken icons. Hope this could help the devtools team to fix this or provide an option to **disable icons completely** or **disable extensions icons only**.

In the current state, using vertical layout in the activity bar is somewhat annoying, as I have to either remember or hover over those blank spaces. Having at least a first letter or the ability to set my own icon/text for the extension would really help.

<img width="1105" height="827" alt="Image" src="https://github.com/user-attachments/assets/0fe2c584-28ca-4a81-836d-f4663288b4cf" />


**Extensions with working icons:**

* [FQM](https://chromewebstore.google.com/detail/feature-queries-manager/fbhgnconlfgmienbmpbeeenffagggonp)
* [CSS Feature Toggles](https://chromewebstore.google.com/detail/css-feature-toggles/aeinmfddnniiloadoappmdnffcbffnjg)

**Extensions with broken icons:**
* [Testing playground](https://chromewebstore.google.com/detail/testing-playground/hejbmebodbijjdhflfknehhcgaklhano).
* [webhint](https://microsoftedge.microsoft.com/addons/detail/webhint/mlgfbihcfnkaenjpdcngdnhcpkdmcdee)
